cognite/client/_api/diagrams.py

```diff
@@ -203,15 +214,30 @@ def detect(
                 batch = items[(self._DETECT_API_FILE_LIMIT * i) : self._DETECT_API_FILE_LIMIT * (i + 1)]
 
                 try:
-                    posted_job = self._run_job(
-                        job_path="/detect",
-                        status_path="/detect/",
-                        items=batch,
-                        entities=entities,
-                        partial_match=partial_match,
-                        search_field=search_field,
-                        min_tokens=min_tokens,
-                        job_cls=DiagramDetectResults,
+                    posted_job = (
+                        self._run_job(
+                            job_path="/detect",
+                            status_path="/detect/",
+                            items=batch,
+                            entities=entities,
+                            partial_match=partial_match,
+                            search_field=search_field,
+                            min_tokens=min_tokens,
+                            job_cls=DiagramDetectResults,
+                            pattern_mode=pattern_mode,
+                            configuration=configuration,
+                        )
+                        if beta_parameters
+                        else self._run_job(
+                            job_path="/detect",
+                            status_path="/detect/",
+                            items=batch,
+                            entities=entities,
+                            partial_match=partial_match,
+                            search_field=search_field,
+                            min_tokens=min_tokens,
+                            job_cls=DiagramDetectResults,
+                        )
```

Comment:
Assuming the suggestion above was used, we can now skip if-else:
```py
posted_job = self._run_job(
    job_path="/detect",
    status_path="/detect/",
    items=batch,
    entities=entities,
    partial_match=partial_match,
    search_field=search_field,
    min_tokens=min_tokens,
    job_cls=DiagramDetectResults,
    **beta_parameters,
)
```

Reply:
I tried to make this work first, but did not find a way to make mypy happy with it.

Reply:
When we can halve the number of codelines in an easy-to-read fashion, I'd much rather have a type ignore comment (personal preference maybe 😅 )

Reply:
type-ignored now